cli/cmd/initialize.go

```diff
@@ -41,39 +48,51 @@ func handleTLSOptions() {
 			},
 		}
 		_ = survey.AskOne(modePrompt, &tlsMode)
+	} else {
+		tlsMode = Import
+	}
 
-		if tlsMode == Generate {
-			// Generate mode requires a host entry
-			prompt := &survey.Input{
-				Message: "Enter a host DNS entry or IP Address for the cluster ingress",
-			}
-			_ = survey.AskOne(prompt, &initOptions.PKI.Host, survey.WithValidator(survey.Required))
-		} else {
-			// Import mode requires the public and private key paths
-			prompt := &survey.Input{
-				Message: "Enter a file path to the ingress public key",
-				Suggest: func(toComplete string) []string {
-					// Give some suggestions to users
-					files, _ := filepath.Glob(toComplete + "*")
-					return files
-				},
-			}
-			_ = survey.AskOne(prompt, &initOptions.PKI.CertPublicPath, survey.WithValidator(survey.Required))
+	// Always ask for a host entry to avoid having to guess which entry in a cert if provided
+	prompt := &survey.Input{
+		Message: "Enter a host DNS entry or IP Address for the cluster ingress. If using localhost, use 127.0.0.1",
+	}
+	_ = survey.AskOne(prompt, &state.TLS.Host, survey.WithValidator(survey.Required))
 
-			prompt.Message = "Enter a file path to the ingress private key"
-			_ = survey.AskOne(prompt, &initOptions.PKI.CertPrivatePath, survey.WithValidator(survey.Required))
+	if tlsMode != Generate {
+		// Import mode requires the public and private key paths
+		prompt := &survey.Input{
+			Message: "Enter a file path to the ingress public key",
+			Suggest: func(toComplete string) []string {
+				// Give some suggestions to users
+				files, _ := filepath.Glob(toComplete + "*")
+				return files
+			},
 		}
+		_ = survey.AskOne(prompt, &state.TLS.CertPublicPath, survey.WithValidator(survey.Required))
+
+		prompt.Message = "Enter a file path to the ingress private key"
+		_ = survey.AskOne(prompt, &state.TLS.CertPrivatePath, survey.WithValidator(survey.Required))
 	}
-	if !utils.CheckHostName(initOptions.PKI.Host) {
-		logrus.Fatalf("The hostname provided (%v) was not a valid hostname. The hostname can only contain: 'a-z', 'A-Z', '0-9', '-', and '.' characters.\n", initOptions.PKI.Host)
+
+	if !utils.CheckHostName(state.TLS.Host) {
+		// On error warn user and cycle the function
+		logrus.Warnf("The hostname provided (%v) was not a valid hostname. The hostname can only contain: 'a-z', 'A-Z', '0-9', '-', and '.' characters as defined by RFC-1035.  If using localhost, you must use the 127.0.0.1.\n", state.TLS.Host)
+		handleTLSOptions()
```

Comment:
I think `survey.AskOne()` interprets `ctrl-C` as a character input (or it just stops the AskOne process instead of killing the parent command?). Because we have this cursive call a user can get stuck trying to exit this command leading to spamming of `ctrl-C` without actually killing the command.

We should (maybe not in this PR but make a ticket) check what output `survey.AskOne()` generated when a user uses `ctrl-C` and make sure we kill the process if we receive that.

Reply:
yeah I noticed that too--super annoying

Reply:
TIL. Cool!

Reply:
fixed in the last push